### ISSUE TITLE
LoRA adapter library: per-agent adapters with hot-swap via Lemonade Server

### ISSUE THREAD

**Issue by kovtcharov** [amd/gaia#668]
## Summary

Create a library of task-specific LoRA adapters that swap at inference time via Lemonade Server. Each agent type (chat, code, email, calendar, finance) gets its own adapter fine-tuned for that vertical.

## Why LoRA Adapters

- Train only 0.1-1% of parameters → recovers 90-95% of full fine-tuning quality
- A single base model + multiple small adapters (10-100MB each) vs. multiple full models (4-16GB each)
- Hot-swap at inference: switch from "email mode" to "code mode" without reloading the base model
- Users can create custom adapters for their specific workflows

## Architecture

```
Lemonade Server
  └── Base Model (Qwen3-8B-GGUF)
       ├── chat_adapter.gguf    (general conversation)
       ├── code_adapter.gguf    (code generation)
       ├── email_adapter.gguf   (email triage + drafting)
       ├── calendar_adapter.gguf (scheduling)
       ├── finance_adapter.gguf (financial analysis)
       └── custom_adapter.gguf  (user-created)
```

## Pre-built Adapters (ship with GAIA)

| Adapter | Trained On | Improves |
|---|---|---|
| Tool Calling | GAIA eval ground truth | Correct tool selection + parameters |
| Code Generation | Code task completions | Coding accuracy + style |
| Email/Communication | Email drafting patterns | Professional tone + format |
| Structured Extraction | Form/table parsing | Data extraction accuracy |

## Dependencies

- Lemonade Server LoRA support (may need Lemonade v10 feature)
- Unsloth integration (for training adapters)
- Eval-to-training pipeline (for training data)

## Acceptance Criteria

- [ ] LoRA adapters load/unload in Lemonade Server without base model reload
- [ ] Agent registry selects appropriate adapter per agent type
- [ ] Pre-built adapters improve eval scores vs. base model
- [ ] Users can create and register custom adapters
- [ ] `gaia finetune --export adapter.gguf` produces Lemonade-compatible adapter
